### PR TITLE
PyTest integration tests prototype

### PR DESCRIPTION
**Pull request by Stringy**
## Description

This contains some experiments with pytest conversion of the existing integration tests.

### CODE REVIEW

File: integration-tests/python/conftest.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    collector = runtime.run(
+        collector_image,
+        name="collector",
+        privileged=True,
+        network="host",
+        mounts=list(mounts.values()),
+        env=env,
+    )
+
+    time.sleep(30)
```
Comment (Molter73) [inline integration-tests/python/conftest.py:104]: 30 seconds?! We really need to just add some testing probe that we can query using http

File: integration-tests/python/conftest.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    yield collector
+
+    try:
+        collector.kill()
+    except Exception:
+        pass
```
Comment (Molter73) [inline integration-tests/python/conftest.py:111]: What if we fail to kill collector before the next test tries to create it again?

**Review by Molter73 — COMMENTED** [review]
This is awesome ❤️

File: integration-tests/python/common/events.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                if process_bucket:
+                    container_bucket = process_bucket.bucket(container.encode())
+                    if container_bucket:
+                        return
+                    
+            time.sleep(1)
+
+        raise RuntimeError(f'Timed out waiting for processes for {container}')
+
+    def processes(self, container=None):
```
Comment (JoukoVirtanen) [inline integration-tests/python/common/events.py:38]: Use a more descriptive function name. Also add a comment explaining what the function does.
```suggestion
    # Gets processes from BoltDB. If container is not set all processes will be returned.
    # If container is set, only processes for that container will be returned.
    def processes(self, container=None):
```

File: integration-tests/python/common/events.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        pass
+
+class BoltDBEvents(Events):
+    def __init__(self):
+        # lazy load the DB on demand, to avoid
+        # weird behavior when opening the DB really 
+        # early in the test
+        self.db = None
+
+    def await_processes(self, container, timeout=30):
```
Comment (JoukoVirtanen) [inline integration-tests/python/common/events.py:25]: Could await_processes return the processes as well.

**Pull request closed without merging**